test/Errors.hs

```diff
@@ -56,54 +52,83 @@ testErrorsColumns TestEnv{..} = TestCase $ do
+    ConversionFailed
+      { errSQLType = "1 values: [(\"INTEGER\",\"SQLInteger 1\")]"
+      , errHaskellType = "at least 2 slots in target type"
+      , errMessage = "mismatch between number of columns to convert and number in target type"
+      }
+  -- Mismatching types (source int,text doesn't match dst string,int)
+  assertResultErrorThrown (query_ conn "SELECT id, t FROM cols" :: IO [(String, Int)])
+    ConversionFailed {errSQLType = "TEXT", errHaskellType = "Int", errMessage = "need an int"}
+  -- Mismatching types (source string doesn't match dst integer)
+  assertResultErrorThrown (query_ conn "SELECT 'foo'" :: IO [Only Integer])
+    ConversionFailed {errSQLType = "TEXT", errHaskellType = "Integer", errMessage = "need an int"}
   -- Mismatching types (sources don't match destination float/double type)
-  assertResultErrorCaught (query_ conn "SELECT 1" :: IO [Only Double])
-  assertResultErrorCaught (query_ conn "SELECT 'foo'" :: IO [Only Double])
-  assertResultErrorCaught (query_ conn "SELECT 1" :: IO [Only Float])
-  assertResultErrorCaught (query_ conn "SELECT 'foo'" :: IO [Only Float])
+  assertResultErrorThrown (query_ conn "SELECT 1" :: IO [Only Double])
+    ConversionFailed {errSQLType = "INTEGER", errHaskellType = "Double", errMessage = "expecting an SQLFloat column type"}
+  assertResultErrorThrown (query_ conn "SELECT 'foo'" :: IO [Only Double])
+    ConversionFailed {errSQLType = "TEXT", errHaskellType = "Double", errMessage = "expecting an SQLFloat column type"}
+  assertResultErrorThrown (query_ conn "SELECT 1" :: IO [Only Float])
+    ConversionFailed {errSQLType = "INTEGER", errHaskellType = "Float", errMessage = "expecting an SQLFloat column type"}
+  assertResultErrorThrown (query_ conn "SELECT 'foo'" :: IO [Only Float])
+    ConversionFailed {errSQLType = "TEXT", errHaskellType = "Float", errMessage = "expecting an SQLFloat column type"}
   -- Mismatching types (sources don't match destination bool type, or is out of bounds)
-  assertResultErrorCaught (query_ conn "SELECT 'true'" :: IO [Only Bool])
-  assertResultErrorCaught (query_ conn "SELECT 2" :: IO [Only Bool])
+  assertResultErrorThrown (query_ conn "SELECT 'true'" :: IO [Only Bool])
+    ConversionFailed {errSQLType = "TEXT", errHaskellType = "Bool", errMessage = "expecting an SQLInteger column type"}
+  assertResultErrorThrown (query_ conn "SELECT 2" :: IO [Only Bool])
+    ConversionFailed {errSQLType = "INTEGER", errHaskellType = "Bool", errMessage = "bool must be 0 or 1, got 2"}
   -- Mismatching types (sources don't match destination string types (text, string)
-  assertResultErrorCaught (query_ conn "SELECT 1" :: IO [Only T.Text])
-  assertResultErrorCaught (query_ conn "SELECT 1" :: IO [Only LT.Text])
-  assertResultErrorCaught (query_ conn "SELECT 1.0" :: IO [Only T.Text])
-  assertResultErrorCaught (query_ conn "SELECT 1.0" :: IO [Only LT.Text])
+  {- TODO these 4 tests are NOT actually throwing any error
```

Comment:
I commented out these 4 tests, because they are not actually throwing any error - they correctly parse the results as "1" and "1.0".

This is also happening on master (it's not bug introduced by changes in this PR). Should this test be removed or should the library behavior be changed so that errors are thrown?

Reply:
I have reverted the commit responsible for the test failure. (See #98 & #99). Did you make these test changes just for the failure or were you also trying to improve the test or test code in other ways?